File: backend/src/main/java/com/project/gameHubBackend/domain/Game.java
package com.project.gameHubBackend.domain;


import lombok.*;

import java.util.Set;


@With
@Value
@Builder
@EqualsAndHashCode(of = {"name","mainPhoto","description","gamePlatforms"})
@ToString(of = {"gameId", "gameNumber", "name"})
public class Game {

     Integer gameId;
     String gameNumber;
     String name;
     String description;
     String mainPhoto;
     Boolean isContinuouslyDelivered;
     Boolean isSoldOut;
     Trailer trailer;
     Set<Photo> photos;
     Set<PublisherGame> publishers;
     Set<CategoryGame> gameCategories;
     Set<PlatformGame> gamePlatforms;
     Set<OpinionCustomerGame> gameOpinions;
}

File: frontend/src/entities/Game.ts
import Category from "./Category";
import Platform from "./Platform";





export default interface Game {
  gameId: number;
  gameNumber: string;
  name: string;
  mainPhoto: string;
  trailerPreviewImage: string;
  trailerUrl:string;
  platforms:string[];
  price:string;
}
